src/modules/Event/page.tsx

```diff
@@ -72,218 +109,377 @@ export function Event() {
+                                      <td className="px-[10px] py-[6px] align-top font-medium text-foreground-300">
+                                        {index}
+                                      </td>
+                                      <td className="px-[10px] py-[6px] font-mono break-all">
+                                        {renderValue(value, format)}
+                                      </td>
+                                    </tr>
+                                  );
+                                },
+                              )}
+                            </tbody>
+                          </table>
+                        </div>
+                      ) : (
+                        <p className="text-[12px] text-foreground-300">
+                          No raw data found.
+                        </p>
+                      )}
+                    </section>
                   </div>
-                </div>
+                </TabsContent>
 
-                <div className="overflow-x-auto">
-                  <table className="w-full table-fixed">
-                    <thead>
-                      <tr className="bg-[#D9D9D9] text-black">
-                        <th className="px-4 py-2 text-left">Input</th>
-                        <th className="px-4 py-2 text-left">Type</th>
-                        <th className="px-4 py-2 text-left">Data</th>
-                      </tr>
-                    </thead>
-                    <tbody>
-                      {eventKeys?.map((key, index) => {
-                        const format = formats.keys ?? "hex";
-                        return (
-                          <tr
-                            key={index}
-                            className="border-b border-dashed last:border-b-0"
-                          >
-                            <td className="px-4 py-2 overflow-hidden text-left">
-                              {key.input}
-                            </td>
-                            <td className="px-4 py-2 break-all">
-                              {key.input_type}
-                            </td>
-                            <td className="px-4 py-2 break-all overflow-x-auto">
-                              <div className="overflow-x-auto">
-                                {convertValue(key.data)?.[format] ?? key.data}
-                              </div>
-                            </td>
-                          </tr>
-                        );
-                      })}
-                    </tbody>
-                  </table>
-                </div>
-              </div>
-            ) : null}
+                {hasDecoded ? (
+                  <TabsContent
+                    value="decoded"
+                    className="mt-0 flex-1 overflow-auto"
+                  >
+                    {/* TODO: handle the nested events without flat, where the selector could be composed of multiple felts. */}
```

Comment:
This is not critical, but definitely worthy to have a look later. But most of contracts are using `flat` attribute.